src/Orleans.Serialization/Codecs/FloatCodec.cs

```diff
@@ -258,7 +277,131 @@ private struct DecimalConverter
+        private static readonly Type CodecFieldType = typeof(Half);
+
+        /// <inheritdoc/>
+        void IFieldCodec<Half>.WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer,
+            uint fieldIdDelta,
+            Type expectedType,
+            Half value) => WriteField(ref writer, fieldIdDelta, expectedType, value);
+
+        /// <summary>
+        /// Writes a field.
+        /// </summary>
+        /// <typeparam name="TBufferWriter">The buffer writer type.</typeparam>
+        /// <param name="writer">The writer.</param>
+        /// <param name="fieldIdDelta">The field identifier delta.</param>
+        /// <param name="expectedType">The expected type.</param>
+        /// <param name="value">The value.</param>
+        public static void WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, Half value) where TBufferWriter : IBufferWriter<byte>
+        {
+            var asUShort = BitConverter.HalfToUInt16Bits(value);
+            UInt16Codec.WriteField(ref writer, fieldIdDelta, expectedType, asUShort, typeof(Half));
+        }
+
+        /// <inheritdoc/>
+        Half IFieldCodec<Half>.ReadValue<TInput>(ref Reader<TInput> reader, Field field) => ReadValue(ref reader, field);
+
+        /// <summary>
+        /// Reads a value.
+        /// </summary>
+        /// <typeparam name="TInput">The reader input type.</typeparam>
+        /// <param name="reader">The reader.</param>
+        /// <param name="field">The field.</param>
+        /// <returns>The value.</returns>
+        public static Half ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+        {
+            ReferenceCodec.MarkValueField(reader.Session);
+            switch (field.WireType)
+            {
+                case WireType.VarInt:
+                    return ReadHalfRaw(ref reader);
+                case WireType.Fixed32:
+                    {
+                        var value = FloatCodec.ReadFloatRaw(ref reader);
+                        if (value > (float)Half.MaxValue || value < (float)Half.MinValue && !float.IsInfinity(value) && !float.IsNaN(value))
+                        {
+                            ThrowValueOutOfRange(value);
+                        }
+
+                        return (Half)value;
+                    }
+                case WireType.Fixed64:
+                    {
+                        var value = DoubleCodec.ReadDoubleRaw(ref reader);
+                        if ((value > (double)Half.MaxValue || value < (double)Half.MinValue) && !double.IsInfinity(value) && !double.IsNaN(value))
+                        {
+                            ThrowValueOutOfRange(value);
+                        }
+
+                        return (Half)value;
+                    }
+
+                case WireType.LengthPrefixed:
+                    {
+                        var width = reader.ReadVarUInt32();
+                        if (width == Width)
```

Comment:
Why bother with supporting this length-prefixed 2-byte version? It's unlikely to be an actual floating point value since no codec will encode a `Half` like that.

Reply:
Fixed